GameOfRunes/GameOfRunes/GameEngine/Entities/EnemyEntity.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  EnemyEntity.swift
+//  GameOfRunes
+//
+//  Created by Jermy on 8/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import SpriteKit
+import GameplayKit
+
+class EnemyEntity: GKEntity {
+    private (set) var gestureEntities: [GestureEntity] = []
+    private unowned var gameEngine: GameEngine
+    
+    init(enemyType: EnemyType, gameEngine: GameEngine) {
+        self.gameEngine = gameEngine
+
+        super.init()
+
+        let enemyAtlas = SKTextureAtlas(named: enemyType.rawValue)
+        let spriteComponent = SpriteComponent(texture: enemyAtlas.textureNamed("WALK_002"))
+        let teamComponent = TeamComponent(team: .enemy)
+        let healthComponent = HealthComponent(healthPoints: 1)
+
+        let moveComponent = MoveComponent(
+            maxSpeed: 150.0,
+            maxAcceleration: 5.0,
+            radius: .init(spriteComponent.node.size.width) * 0.01,
+            gameEngine: gameEngine
+        )
+
+        addComponent(spriteComponent)
+        addComponent(teamComponent)
+        addComponent(moveComponent)
```

Comment:
I think the addComponent(moveComponent) line is redundant? Since we are calling addMoveComponent below.
Also, maybe we shouldn't duplicate the code? Like right now we are creating a MoveComponent at line 26 and at line 68.